repair/repair.cc

```diff
@@ -76,14 +76,16 @@ std::ostream& operator<<(std::ostream& out, row_level_diff_detect_algorithm algo
     return out << "unknown";
 }
 
-static std::vector<sstring> list_column_families(const database& db, const sstring& keyspace) {
+static std::pair<std::vector<sstring>, std::vector<schema_ptr>> list_column_families(const database& db, const sstring& keyspace) {
     std::vector<sstring> ret;
```

Comment:
schema contains the name, so it's enough to return the schema vector.

Reply:
You're obviously right. The reason I decided to return both is to minimize the change.

list_column_families is used in line 1415. In there you can see that code following the line works on a vector of column family names. There are two cases those names come from: from options (line 1401/1406) or from list_column_families (line 1415). To unify the two cases I decided to:
1. In the first option when cfs come from options, I generate vector of schema_ptrs from names
2. In the second option I generate both vector of names and vector of schema_ptrs in list_column_families
This way rest of the code can work on those two vectors no matter which case it is.
There is an alternative to return just a vector of schema_ptrs and change all the rest of the code to use it instead of the vector of names (cfs) but that is much bigger change and I wanted to minimize the friction.
@avikivity and @nyh do you want me to change all that code so that we can return single vector of schemas here?

Reply:
With a check being moved to repair_info, change to list_column_families will be reverted so we can consider this commend addressed.